packages/app-builder/src/components/ui/Toolbar/toolbar-item.tsx

```diff
@@ -0,0 +1,89 @@
+import { useNode } from '@craftjs/core'
+import { elMb2, Label } from '@reapit/elements'
+import React, { FC, ReactNodeArray } from 'react'
+
+import { ToolbarDropdown } from './toolbar-dropdown'
+import { ToolbarTextInput } from './toolbar-text-input'
+import { ToolbarItemType } from './types'
+
+export type ToolbarItemProps = {
+  prefix?: string
+  label?: string
+  full?: boolean
+  propKey: string
+  index?: number
+  children?: ReactNodeArray
+  type: ToolbarItemType
+  title?: string
+}
+
+const ToolbarItemInput: FC<ToolbarItemProps> = ({ propKey, type, index = 0, ...props }: ToolbarItemProps) => {
+  const {
+    actions: { setProp },
+    propValue,
+  } = useNode((node) => ({
+    propValue: node.data.props[propKey],
+  }))
+  const isArray = Array.isArray(propValue)
+  const value = isArray ? propValue[index] : propValue
+
+  switch (type) {
+    case ToolbarItemType.Text:
+    case ToolbarItemType.Number:
+      return (
+        <ToolbarTextInput
+          {...props}
+          type={type}
+          value={value}
+          onChange={(value) => {
+            setProp((props: any) => {
+              if (type === ToolbarItemType.Number) {
+                value = parseInt(value)
+              }
+              if (isArray) {
+                props[propKey][index] = value
+              } else {
+                props[propKey] = value
+              }
+            }, 500)
+          }}
+        />
+      )
+    case ToolbarItemType.Radio:
+      return (
+        <>
+          {props.label ? <Label>{props.label}</Label> : null}
+          {React.Children.map(props.children, (child: React.ReactNode) => {
+            React.isValidElement(child) &&
+              React.cloneElement(child, {
+                onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+                  const { value, checked } = e.currentTarget
+                  if (checked) {
+                    setProp((props) => {
+                      props[propKey] = value
+                    })
+                  }
+                },
+              })
+          })}
+        </>
+      )
+    case ToolbarItemType.Select:
+      return props.title ? (
+        <ToolbarDropdown
+          value={value || ''}
+          onChange={(value) => setProp((props) => (props[propKey] = value))}
+          title={props.title}
+          {...props}
+        />
+      ) : null
+    default:
+      return null
+  }
```

Comment:
shouldn't need default as type is always set no?

Reply:
I think TS/ESLint will not like the switch if you don't offer a default

Reply:
yeahh it's not smart enough to figure it out from the type sadly